internal/cli/streams/privatelink/create.go

```diff
@@ -0,0 +1,110 @@
+// Copyright 2025 MongoDB Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package privatelink
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"github.com/mongodb/mongodb-atlas-cli/atlascli/internal/cli"
+	"github.com/mongodb/mongodb-atlas-cli/atlascli/internal/cli/require"
+	"github.com/mongodb/mongodb-atlas-cli/atlascli/internal/config"
+	"github.com/mongodb/mongodb-atlas-cli/atlascli/internal/file"
+	"github.com/mongodb/mongodb-atlas-cli/atlascli/internal/flag"
+	"github.com/mongodb/mongodb-atlas-cli/atlascli/internal/store"
+	"github.com/mongodb/mongodb-atlas-cli/atlascli/internal/usage"
+	"github.com/spf13/afero"
+	"github.com/spf13/cobra"
+	atlasv2 "go.mongodb.org/atlas-sdk/v20241113004/admin"
+)
+
+var createTemplate = "Atlas Stream Processing PrivateLink endpoint {{.InterfaceEndpointId}} created.\n"
+
+type CreateOpts struct {
+	cli.ProjectOpts
+	cli.OutputOpts
+	store    store.PrivateLinkCreator
+	filename string
+	fs       afero.Fs
+}
+
+func (opts *CreateOpts) Run() error {
+	privateLinkEndpoint := atlasv2.NewStreamsPrivateLinkConnection()
+	if err := file.Load(opts.fs, opts.filename, privateLinkEndpoint); err != nil {
+		return err
+	}
+
+	// Remaining validation will be done by the API
+	if len(privateLinkEndpoint.GetProvider()) == 0 {
```

Comment:
this should be safe now that you use getters
```suggestion
	if privateLinkEndpoint.GetProvider() == "" {
```